### DISCUSSION TITLE
NRF24 "not configured" and chip type "unknown" after upgrade

### DISCUSSION THREAD

**Discussion by PHIL-IP** [tbnobody/OpenDTU#1295] [Q&A] [answered]
Hi all,
I had an openDTU that ran for several months without a problem under an older version. After completely erasing the ESP32 with esptool, I put back the latest version of openDTU. Now, in the system information, the NRF24 is no longer "seen", it is marked as "not configured" and its type is "unknown". I don't see why, as it was well recognized in the old version. 
Note also that the openDTU version in the information is 0000000! 
Am I missing something?
![14-IMAGE](https://github.com/tbnobody/OpenDTU/assets/26251016/0e9c3dbd-ac64-43c2-a77a-ce206c184013)

**Comment by tbnobody** [comment]
Did you refresh the web UI properly? (CTRL+F5)
Did you upload a device profile? https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles

**Comment by PHIL-IP** [comment]
Thanks for your reply.
Yes refresh has no effect and I don't upload any device profile because all the hardware is the same that for older version.

**Reply by tbnobody** [reply]
> I don't upload any device profile because all the hardware is the same that for older version.

But it seems that you've uploaded a different environment then previously... see https://github.com/tbnobody/OpenDTU/wiki/Binary-Installation

**Reply by PHIL-IP** [reply]
I've used the same code for 3 identical DevKits and uploaded it several times using VSCode. 
I'm sorry, but I just don't understand anything anymore. :( 
Did I forget to do something before reflashing the code?
No matter what I do, the NRF2401 always remains "not configured". :(

**Reply by tbnobody** [reply]
If you did not upload any device profile in the past (and you compiled the code yourself) you maybe have to choose the deprecated `generic`"environment instead of the `generic_esp32` environment.

> Did I forget to do something before reflashing the code?

You didn't forget anything. You just have to upload a device profile (or use the old environment). That's called "new features".... Code is changing...

**Reply by PHIL-IP** [reply]
MANY THANKS FOR YOUR PATIENCE
I have trying to upload 'nodemcu_esp32.json' and now NRF2401 works. 👍 
![15-IMAGE](https://github.com/tbnobody/OpenDTU/assets/26251016/a8fa67d1-2577-4f2a-9869-3317b5808f9e)

I also have an over problem, inverters not responding :( 
This is 2 HM350 3 meters from the openDTU
Note that these 2 inverters have already sent me their data under the old version of openDTU (I've double-checked their serial number and it's ok).

In the console I have this :
`RX Period End
13:52:10.228 > All missing
13:52:10.228 > Nothing received, resend whole request
13:52:10.228 > TX SystemConfigPara Channel: 40 --> 15 73 30 95 96 80 13 06 76 80 05 00 64 F1 D0 64 00 00 00 00 00 00 00 00 DF 46 8B 
13:52:10.485 > RX Period End
13:52:10.485 > All missing
13:52:10.485 > Nothing received, resend whole request
13:52:10.485 > TX SystemConfigPara Channel: 61 --> 15 73 30 95 96 80 13 06 76 80 05 00 64 F1 D0 64 00 00 00 00 00 00 00 00 DF 46 8B 
13:52:10.730 > RX Period End
13:52:10.730 > All missing
13:52:10.730 > Nothing received, resend whole request
13:52:10.730 > TX SystemConfigPara Channel: 75 --> 15 73 30 95 96 80 13 06 76 80 05 00 64 F1 D0 64 00 00 00 00 00 00 00 00 DF 46 8B 
13:52:10.973 > RX Period End
13:52:10.973 > All missing
13:52:10.973 > Nothing received, resend whole request
13:52:10.973 > TX SystemConfigPara Channel: 3 --> 15 73 30 95 96 80 13 06 76 80 05 00 64 F1 D0 64 00 00 00 00 00 00 00 00 DF 46 8B 
13:52:11.380 > RX Period End
13:52:11.380 > All missing
13:52:11.380 > Nothing received, resend count exeeded
13:52:11.380 > Fetch inverter: 112173309888
13:52:11.380 > Request SystemConfigPara
13:52:11.438 > TX RealTimeRunData Channel: 23 --> 15 73 30 98 88 80 13 06 76 80 0B 00 64 F1 D0 6C 00 00 00 00 00 00 00 00 D1 2E F8 
13:52:11.779 > RX Period End
13:52:11.779 > All missing
13:52:11.779 > Nothing received, resend whole request
13:52:11.779 > TX RealTimeRunData Channel: 40 --> 15 73 30 98 88 80 13 06 76 80 0B 00 64 F1 D0 6C 00 00 00 00 00 00 00 00 D1 2E F8 
13:52:12.353 > RX Period End
13:52:12.353 > All missing
13:52:12.353 > Nothing received, resend whole request
13:52:12.353 > TX RealTimeRunData Channel: 61 --> 15 73 30 98 88 80 13 06 76 80 0B 00 64 F1 D0 6C 00 00 00 00 00 00 00 00 D1 2E F8 
13:52:12.892 > RX Period End
13:52:12.892 > All missing
13:52:12.892 > Nothing received, resend whole request
13:52:12.892 > TX RealTimeRunData Channel: 75 --> 15 73 30 98 88 80 13 06 76 80 0B 00 64 F1 D0 6C 00 00 00 00 00 00 00 00 D1 2E F8 
13:52:13.431 > RX Period End
13:52:13.431 > All missing
13:52:13.431 > Nothing received, resend whole request
13:52:13.431 > TX RealTimeRunData Channel: 3 --> 15 73 30 98 88 80 13 06 76 80 0B 00 64 F1 D0 6C 00 00 00 00 00 00 00 00 D1 2E F8 
13:52:18.099 > RX Period End
13:52:18.099 > All missing
13:52:18.099 > Nothing received, resend count exeeded
13:52:18.109 > TX AlarmData Channel: 23 --> 15 73 30 98 88 80 13 06 76 80 11 00 64 F1 D0 6C 00 00 00 00 00 00 00 00 0B 35 23 
`

**Reply by tbnobody** [reply, marked as answer]
First of all you should double check whether the IRQ pin in your selected device profile (settings --> hardware) matches your connection to the NRF module

**Reply by PHIL-IP** [reply]
I've checked the NRF2104 connections on the board and the settings in the json files at least 3 times and everything matches up well.

**Reply by PHIL-IP** [reply]
I have rebuilded a brand new DTU board and now all is works fine !!!!!

AGAIN THANKS FOR YOUR PATIENCE

**Comment by PHIL-IP** [comment]
ALL IS WORKS FINE NOW.
THANKS A LOT FOR ALL YOUR HELP.

Best regards

**Comment by lupyn10** [comment]
HI,
I have the same problem. but I can't solve it.
I attach some photos with esp-wroom-32 at 30pin.

![Immagine 2023-09-19 185145](https://github.com/tbnobody/OpenDTU/assets/144829845/e227285b-478b-4d02-8be5-e029b8fa85e8)

![Immagine 2023-09-19 185236](https://github.com/tbnobody/OpenDTU/assets/144829845/79c0413f-4033-4fb7-8e82-dbb33f637282)

![20230919_185642](https://github.com/tbnobody/OpenDTU/assets/144829845/115a0231-9e82-43e4-8e82-da0ad677d442)

![20230919_185722](https://github.com/tbnobody/OpenDTU/assets/144829845/bf1b4ceb-56a9-4091-b781-1c1b14e60e85)

![20230919_190551](https://github.com/tbnobody/OpenDTU/assets/144829845/ca4ce312-92ff-42cf-97aa-c1ac0b20f0c2)

**Reply by tbnobody** [reply]
You are using the generic_esp32 firmware. This firmware requires a device profile. See https://github.com/tbnobody/OpenDTU/wiki/Binary-Installation and https://github.com/tbnobody/OpenDTU/wiki/Device-Profiles

**Comment by lupyn10** [comment]
Thanks a lot it works. Someone managed to unlock the 600 to 800 watt limit of a tsun m800de microinverter
